packages/testing/src/chai-retry-plugin/chai-retry-plugin.ts

```diff
@@ -29,67 +30,71 @@ import type { AssertionMethod, FunctionToRetry, AssertionStackItem, RetryOptions
  * await expect(funcToRetry).retry().and.have.property('success').and.be.true;
  * ```
  */
-export const chaiRetryPlugin = function (_: typeof Chai, utils: Chai.ChaiUtils) {
-    Chai.Assertion.addMethod('retry', function (retryOptions: RetryOptions = {}): PromiseLikeAssertion {
-        const functionToRetry: FunctionToRetry = this._obj as FunctionToRetry;
-        const description = utils.flag(this, 'message') as string;
+export const chaiRetryPlugin = function (_: typeof Chai, { flag, inspect }: Chai.ChaiUtils) {
+    Object.defineProperty(Chai.Assertion.prototype, 'retry', {
+        value: function (retryOptions: RetryOptions = {}): PromiseLikeAssertion {
```

Comment:
Extract this to a named function instead of increasing indentation even further.